File: AtCoder/abc202/C.kt
private fun readLn() = readLine()!!
private fun readInt() = readLn().toInt()
private fun readInts() = readLn().split(" ").map { it.toInt() }

fun main(args: Array<String>) {
    val n = readInt()
    val a = readInts()
    val b = readInts()
    val c = readInts()
    val mp = hashMapOf<Int, Int>()
    a.forEach { mp[it] = mp.getOrDefault(it, 0) + 1 }
    var ans = 0L
    c.forEach {
        ans += mp.getOrDefault(b[it - 1], 0).toLong()
    }
    print(ans)
}

File: AtCoder/abc202/C.swift
let n = Int(readLine()!)!
let a = readLine()!.split(separator: " ").map({ Int($0)! })
let b = readLine()!.split(separator: " ").map({ Int($0)! })
let c = readLine()!.split(separator: " ").map({ Int($0)! })
var mp = [Int:Int]()
a.forEach {
	mp[$0] = (mp[$0] ?? 0) + 1
}
var ans = 0
for i in 0..<n {
	let curr = b[c[i] - 1]
	ans += mp[curr] ?? 0
}
print(ans)
